### PR TITLE
Handing partition SA / platform SA

### PR DESCRIPTION
**Pull request by lxwinspur**
Signed-off-by: George Liu <liuxiwei@inspur.com>

### CODE REVIEW

**Comment by santoshpuranik** [comment]
Codewise this looks good to me. I need to get this on a system and test. I plan to do that this week, will approve once my testing looks good.

**Comment by lxwinspur** [comment]
> 

Ack

**Review by jinuthomas — APPROVED** [review]

File: ibm-sai.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    for (const auto& fruInstancePath : endpoints)
+    {
+        // Set OperationalStatus by fru instance path
+        try
+        {
+            utils::PropertyValue functionalValue{value};
+            utils::DBusHandler().setProperty(
+                fruInstancePath,
+                "xyz.openbmc_project.State.Decorator.OperationalStatus",
+                "Functional", functionalValue);
```
Comment (santoshpuranik) [inline ibm-sai.cpp:57]: The use case here is actually slightly different - We have two LED group objects, the PLATFORM_SAI and PARTITION_SAI pointing to the same inventory object (See https://github.com/openbmc/openbmc/blob/a2cc401d586a9b1a3cdf2e93847839b551b48074/meta-ibm/recipes-phosphor/inventory/phosphor-inventory-manager/p10bmc/ibm%2Ceverest_associations.json#L293).

So when setting Operational status to true, we need to sure that both groups are de-asserted.

If we are setting OperationalStatus to true, it is OK to just look at the asserted for the incoming path.

**Review by santoshpuranik — CHANGES_REQUESTED** [review]
Finally got a chance to test this. Some changes are needed based on my testing.

**Comment by santoshpuranik** [comment]
I created #3 by fixing the issues I found when testing this PR. I think we can continue the review there. This can be closed.

File: ibm-sai.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+namespace phosphor
+{
+namespace led
+{
+namespace ibm
+{
+// Set OperationalStatus functional according to the asserted state of the group
+void setOperationalStatus(const std::string& path, bool value)
+{
+    if (path != PARTITION_SAI || path != PLATFORM_SAI)
```
[the remark this answers is not in the document]
Comment (lxwinspur) [reply ibm-sai.cpp]: Done

File: ibm-sai.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        return;
+    }
+
+    // Get endpoints from the rType
+    std::string fru = path + "/fault_led_group";
```
[the remark this answers is not in the document]
Comment (lxwinspur) [reply ibm-sai.cpp]: Done

**Comment by lxwinspur** [comment]
> 

Sure, I will close it

**Pull request closed without merging**